src/modules/commander/Commander.cpp

```diff
@@ -2126,8 +2126,10 @@ Commander::run()
 			if (!_armed.armed) {
 				_status.system_type = _param_mav_type.get();
 
-				const bool is_rotary = is_rotary_wing(_status) || (is_vtol(_status) && _vtol_status.vtol_in_rw_mode);
-				const bool is_fixed = is_fixed_wing(_status) || (is_vtol(_status) && !_vtol_status.vtol_in_rw_mode);
+				const bool is_rotary = is_rotary_wing(_status) || (is_vtol(_status)
+						       && _vtol_status.vehicle_vtol_state != vtol_vehicle_status_s::VEHICLE_VTOL_STATE_FW);
```

Comment:
```suggestion
						       && _vtol_status.vehicle_vtol_state == vtol_vehicle_status_s::VEHICLE_VTOL_STATE_MC);
```

Reply:
No I think it has to be !=FW, as that also includes all the transition states (we want the type to be rotary in transition).

Reply:
@sfuhrer Makes sense